### PR TITLE
cleanup: addresses paralleltest linter

### PR DESCRIPTION
**Pull request by yati1998**
The Go linter paralleltest checks that the t.Parallel
gets called for the test method and for the range of
test cases within the test.

Updates: #2025

Signed-off-by: Yati Padia <ypadia@redhat.com>

### CODE REVIEW

File: internal/util/conn_pool_test.go [hunk 1]
```diff
@@ -74,6 +74,7 @@ func (cp *ConnPool) fakeGet(monitors, user, keyfile string) (*rados.Conn, string
 }
 
 func TestConnPool(t *testing.T) {
+	t.Parallel()
```
Comment (nixpanic) [inline internal/util/conn_pool_test.go:77]: These tests are probably not suitable to run in parallel. After making modifications, try running `go test -v ./internal/util` and see if it fails for you too:
```
=== CONT  TestConnPool/fakeGet
    conn_pool_test.go:97: failed to get connection: could not open keyfile /tmp/conn_utils.keyfile: open /tmp/conn_utils.keyfile: no such file or directory
    conn_pool_test.go:104: there is more than a single conn in cp.conns: 0
    conn_pool_test.go:110: getting the conn from cp.conns failed
--- FAIL: TestConnPool (0.00s)
    --- FAIL: TestConnPool/fakeGet (0.00s)
```

Comment (yati1998) [reply internal/util/conn_pool_test.go:77]: sure

**Review by nixpanic — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e/k8s-1.20

Failed due to #1969 ([logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.20/detail/mini-e2e_k8s-1.20/1277/pipeline)).

**Pull request merged**